### ISSUE TITLE
Add checks for (mis)configurations of CdOd- and Reach-Tasks

### ISSUE THREAD

**Issue by marcellussiegburg** [fmidue/modelling-tasks#138]
Introduce check functionality for configurations to prevent unsupported configurations (before the checker or generator runs into an verification problem or infinite generation loop / abortion error at runtime).

Such checks already exist for the Petri net tasks.

**Comment by jvoigtlaender** [comment]
See https://github.com/jvoigtlaender/modelling-tasks-internal/issues/130#issuecomment-905211338 and the comment(s) following it.

**Comment by jvoigtlaender** [comment]
Constraints like these: https://github.com/jvoigtlaender/modelling-tasks-internal/blob/a8e12a052cd3f2c274838a86011367745449e623/src/Modelling/CdOd/Generate.hs#L67-L70 (currently checked during the search) should then be integrated into the config checkers (and maybe can be deleted where they now appear).

**Comment by marcellussiegburg** [comment]
Especially check constraints for SelectValidCd which is using `rapairIncorrect` and possibly is ending up in never ending loops (e.g. if constraints do not allow removing / adding relationships); see also https://fmidue.slack.com/archives/DP6ACM4S2/p1630058611028000?thread_ts=1630048387.020500&cid=DP6ACM4S2

**Comment by jvoigtlaender** [comment]
So that would be this looping? https://github.com/jvoigtlaender/modelling-tasks-internal/blob/36cf84f966aea8ccf1f06c7d95329d093a780b17/src/Modelling/CdOd/RepairCd.hs#L305 Where Alloy came back with no (useful) instances and thus is started again.

**Comment by marcellussiegburg** [comment]
Yes, i.e. in this case it tries a different set of changes, but if all combinations of change sets are not returning ODs it will loop for ever!

**Comment by marcellussiegburg** [comment]
Instances (i.e. manual configurations) need to be checked as well!
